### PR TITLE
Replacing Guid.NewGuid for ScanId

### PR DESCRIPTION
**Pull request by eddynaka**

### CODE REVIEW

File: Src/Sarif.PatternMatcher/SearchSkimmer.cs [hunk 1]
```diff
@@ -1243,7 +1242,7 @@ private void AddFixToResult(FlexMatch flexMatch, SimpleFix simpleFix, Result res
         {
             int argsCount = namedArgumentToIndexMap.Count;
 
-            var arguments = new List<string>(new string[argsCount]);
```
Comment (eddynaka) [inline Src/Sarif.PatternMatcher/SearchSkimmer.cs:1246]: >new string[argsCount] [](http://example.com/codeflow?start=45&length=21)

this was creating a string array with argsCount position, that wasn't being used.
Passing the count prevents resizing.

File: Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs [hunk 2]
```diff
@@ -15,6 +15,8 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk
 {
     public abstract class ValidatorBase
     {
+        public const string ScanIdentityId = "01350135-0135-4135-0135-013501350135";
```
Comment (michaelcfanning) [inline Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs:18]: >ScanIdentityId [](http://example.com/codeflow?start=28&length=14)

You should make this a property and initialize it with a random GUID value.

File: Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs [hunk 3]
```diff
@@ -53,9 +55,32 @@ protected ValidatorBase()
...
+                }
+
+                return scanIdentityGuid;
+            }
+
+            set
+            {
+                if (!Guid.TryParse(value, out _))
+                {
+                    throw new ArgumentException($"The '{value}' is not a GUID.");
```
Comment (michaelcfanning) [inline Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs:74]: >The '{value}' is not a GUID. [](http://example.com/codeflow?start=50&length=28)

'{value}' is a not a valid GUID.

File: Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs [hunk 4]
```diff
@@ -53,9 +55,32 @@ protected ValidatorBase()
             PerFileFingerprintCache = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
         }
 
+        protected virtual string ScanIdentityGuid
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(scanIdentityGuid))
```
Comment (michaelcfanning) [inline Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs:62]: >IsNullOrEmpty [](http://example.com/codeflow?start=27&length=13)

according to this code, you can never set scanIdentityGuid to an empty value. i.e., a null check here would be just fine. So I'd be tempted to render this code as a two-liner:

scanIdentityGuid ??= Guid.NewGuid().ToString();
return scanIdentityGuid;

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**